src/ahocorasick.rs

```diff
@@ -523,6 +523,17 @@ impl<S: StateID> AhoCorasick<S> {
     ///     true
     /// });
     /// assert_eq!("0 the 2 to the 0age", result);
+    ///
+    /// let mut result = String::new();
+    /// ac.replace_all_with(haystack, &mut result, |mat, _, dst| {
+    ///     dst.push_str(&mat.pattern().to_string());
+    ///     if mat.pattern() == 2 {
+    ///         false
+    ///     } else {
+    ///         true
+    ///     }
```

Comment:
I think you could replace this with `mat.pattern() != 2`.

Reply:
I thought the more expressive version with visible `bool` literals could be better suited for the purpose of the example.